### PR TITLE
Allocate WOLFSSL_X509_NAME with heap hint.

### PR DESCRIPTION
**Pull request by anhu**
# Description

Allow for heap hint to be set when creating a new WOLFSSL_X509_NAME(). Since this is a compat layer API there was previously no concept of a heap in the original API. 

Fixes ZD 16333

# Testing

make check

### CODE REVIEW

File: src/x509.c [hunk 1]
```diff
@@ -9485,20 +9485,24 @@ WOLF_STACK_OF(WOLFSSL_X509)* wolfSSL_X509_chain_up_ref(
...
-    WOLFSSL_X509_NAME* wolfSSL_X509_NAME_new(void)
+    WOLFSSL_X509_NAME* wolfSSL_X509_NAME_new_ex(void *heap)
     {
         WOLFSSL_X509_NAME* name;
 
-        WOLFSSL_ENTER("wolfSSL_X509_NAME_new");
+        WOLFSSL_ENTER("wolfSSL_X509_NAME_new_ex");
 
-        name = (WOLFSSL_X509_NAME*)XMALLOC(sizeof(WOLFSSL_X509_NAME), NULL,
+        name = (WOLFSSL_X509_NAME*)XMALLOC(sizeof(WOLFSSL_X509_NAME), heap,
```
Comment (dgarske) [inline src/x509.c:9494]: The `wolfSSL_X509_NAME_free` needs `name->heap` added on the matching XFREE.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/x509.c (hunk 1 above)
Comment (anhu) [reply src/x509.c:9494]: Nice catch!! Fixing it now.

**Review by dgarske — APPROVED** [review]

**Comment by dgarske** [comment]
Retest this please

**Comment by anhu** [comment]
Jenkins retest this please .

**Comment by anhu** [comment]
Hi David, This is now ready for merging.

**Pull request merged**